<launch>

    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="align_depth" value="true"/>
    </include>

    <node name="object_detection_ssd1" pkg="ros_vino" type="object_detection_ssd" output="screen">
        <param name="topic_image_input"     value="/camera/color/image_raw" />
        <param name="topic_image_output"    value="/camera/image_rects" />
        <param name="score_threshold"       value="0.7" />
        <param name="device"                value="MYRIAD" />
        <param name="model_path"            value="$(find ros_vino)/models/FP16/mobilenet-ssd/mobilenet-ssd.xml" />
        <!-- <param name="model_path"            value="$(find ros_vino)/models/FP16/mobilenet-lb-0/frozen_inference_graph.xml" /> -->
        <param name="bool_auto_resize"      value="true" />
        <param name="bool_pc"               value="true" />
        <param name="bool_raw"              value="false" />
    </node>

    <node name="multi_kf_tracker" pkg="ros_vino" type="multi_kf_tracker">
        <param name="topic_image_input"         value="/camera/color/image_raw" />
        <param name="topic_objects_input"       value="/camera/detected_objects" />
        <param name="topic_image_output"        value="/camera/image_tracked" />
        <param name="topic_objects_output"      value="/camera/tracked_objects" />
        <param name="min_correspondence_cost"   value="100.0" />
        <param name="p_loop_rate"               value="30.0" />
        <param name="p_tracking_duration"       value="0.5" />
        <param name="p_sampling_time"           value="30.0" />
        <param name="p_process_variance"        value="1.0" />
        <param name="p_process_rate_variance"   value="10.0" />
        <param name="p_measurement_variance"    value="10000.0" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find ros_vino)/rviz/demo.rviz" />

</launch>